test/test_heat_storage.jl

```diff
@@ -96,3 +96,113 @@
+            Dict(CO₂ => FixedProfile(0)),   # Emission price for CO₂ in EUR/t
+            CO₂,                            # CO₂ instance
+        )
+
+        nodes = [
+            RefSource(
+                "surplus heat source",      # Node id
+                FixedProfile(0.85),         # Capacity in MW
+                FixedProfile(0),            # Variable OPEX in EUR/MW
+                FixedProfile(0),            # Fixed OPEX in EUR/8h
+                Dict(heat_sur => 1),        # Output from the Node, in this gase, heat_sur
+            ),
+            RefNetworkNode(
+                "HeatPump",
+                FixedProfile(0.5),
+                FixedProfile(0),
+                FixedProfile(0),
+                Dict(heat_sur => 1),
+                Dict(heat_use => 1),
+            ),
+            EMH.FixedRateTES{CyclicStrategic}(
+                "TES",
+                StorCap(FixedProfile(1)),
+                heat_use,
+                0.2,
+                0.25,
+                0.1,
+                Dict(heat_use => 1),
+                Dict(heat_use => 1),
+            ),
+            RefSink(
+                "heat demand",              # Node id
+                OperationalProfile([0.1, 0.1, 0.2, 0.8]), # Demand in MW
+                Dict(:surplus => FixedProfile(0), :deficit => FixedProfile(1e6)),
+                # Line above: Surplus and deficit penalty for the node in EUR/MWh
+                Dict(heat_use => 1),           # Energy demand and corresponding ratio
+            ),
+        ]
+
+        # Connect all nodes with the availability node for the overall energy/mass balance
+        links = [
+            Direct("source-HP", nodes[1], nodes[2], Linear()),
+            Direct("HP-demand", nodes[2], nodes[4], Linear()),
+            Direct("HP-TES", nodes[2], nodes[3], Linear()),
+            Direct("TES-demand", nodes[3], nodes[4], Linear()),
+        ]
+
+        # Input data structure
+        case = Case(T, products, [nodes, links], [[get_nodes, get_links]])
+        return (; case, model, nodes, products, T, op_duration)
+    end
+    case, model, nodes, products, T, op_duration = generate_data()
+    optimizer = optimizer_with_attributes(HiGHS.Optimizer, MOI.Silent() => true)
+    m = run_model(case, model, optimizer)
+
+    heat_use = products[3]
+
+    # Test that the expected heat loss ratio is calculated
+    heatloss = EMH.heat_loss_factor(nodes[3])
+
+    heat_input = sum(JuMP.value(m[:flow_in][nodes[3], t, heat_use]) for t ∈ T) * op_duration
+    heat_output =
+        sum(JuMP.value(m[:flow_out][nodes[3], t, heat_use]) for t ∈ T) * op_duration
+    heat_stored = sum(JuMP.value(m[:stor_level][nodes[3], t]) for t ∈ T)
+
+    calculated_max_charge_rate =
+        maximum([JuMP.value(m[:stor_charge_use][nodes[3], t]) for t ∈ collect(T)])
+    calculated_max_discharge_rate =
+        maximum([JuMP.value(m[:stor_discharge_use][nodes[3], t]) for t ∈ collect(T)])
```

Comment:
```suggestion
    calculated_max_charge_rate =
        maximum([JuMP.value(m[:stor_charge_use][nodes[3], t]) for t ∈ T])
    calculated_max_discharge_rate =
        maximum([JuMP.value(m[:stor_discharge_use][nodes[3], t]) for t ∈ T])
```
The collect is not necessary. If it is, then we have other problems.